qutip/tests/test_superop_reps.py

```diff
@@ -41,76 +41,88 @@
 
 from numpy import abs, pi, asarray, kron
 from numpy.linalg import norm
-from numpy.testing import assert_, assert_almost_equal, run_module_suite, assert_equal
-
-from unittest import expectedFailure
 
+import pytest
+from qutip import composite
 from qutip.qobj import Qobj
 from qutip.states import basis
-from qutip.operators import identity, sigmax, sigmay, qeye, create
+from qutip.operators import identity, sigmax, sigmay, sigmaz, qeye, create
 from qutip.qip.operations.gates import swap
 from qutip.random_objects import rand_super, rand_super_bcsz, rand_dm_ginibre
-from qutip.tensor import tensor, super_tensor
+from qutip.tensor import tensor, super_tensor, tensor_contract
 from qutip.superop_reps import (kraus_to_choi, to_super, to_choi, to_kraus,
                                 to_chi, to_stinespring)
 from qutip.superoperator import operator_to_vector, vector_to_operator, sprepost
 
 tol = 1e-9
 
 
-class TestSuperopReps(object):
+@pytest.fixture(scope="function", params=[2, 3, 7])
+def dimension(request):
+    # There are also some cases in the file where this fixture is explicitly
+    # overridden by a more local mark.  That is deliberate.
+    return request.param
+
+
+@pytest.fixture(scope="function", params=[
+    pytest.param(rand_super, id="super")
+])
+def superoperator(request, dimension):
+    return request.param(dimension)
```

Comment:
This is quite a strange function to read given that it could have just been:
```
@pytest.fixture
def superoperator(dimension):
    return rand_super(dimension)
```
Maybe there was a plan to do something more? Could we either add another param or use the simpler implementation?

Reply:
I think that the idea was to draw from alternative superop distributions like rand rand_super_bcsz, rand_dm_ginibre . Do you think this would be nice to make tests more thorough, or that it just adds time and is not increasing coverage substantially?

Reply:
I think adding the alternative distributions is great if the individual tests are not too slow.